packages/op-coreutils/src/op_coreutils/bigquery/write.py

```diff
@@ -153,25 +157,10 @@ def overwrite_partitions_dynamic(
         expiration_days (int, optional): If provided sets the expiration time of the table.
     """
 
-    # Ensure "dt" is a DateTime
```

Comment:
The logic we were applying to the dataframe here, checking it has `dt` and breaking it into `dt` parts is made into a function so we can reuse it in the `upsert_partitioned_dataset` function